### PR TITLE
Fix page reload

### PR DESCRIPTION
**Pull request by d-spiridonov**
Background:
We need a way to reload the first page in the notes table after saving/editing/deleting a note. The current implementation of smoothPageTable didn't have an option to reload the first page.

**What does this PR do?**
Adds a `reloadFirstPage` function to `smoothPagedTable` which reloads the first page, clears the `rows` state and sets row data to the `rows` first index.

### CODE REVIEW

File: src/table/pagedTable/smoothPagedTable.tsx [hunk 1]
```diff
@@ -105,7 +105,7 @@ export function useSmoothPagedTable<T>(props: IUsePagedTableProps<T>) {
 
       // set newPage as current page
       // someone maybe scrolling like crazy so let's always remember last page
-
+      console.log('res.page', res.page)
```
Comment (d-spiridonov) [inline src/table/pagedTable/smoothPagedTable.tsx:108]: console

File: src/table/pagedTable/models.ts [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import { IPaginatedRequest, IPaginatedResponse } from '../../apiClient/paginated'
 
 export interface IPagedDataContext {
-  reloadData: () => void
+  reloadData: (onPage?: number) => void
```
Comment (d-spiridonov) [inline src/table/pagedTable/models.ts:4]: this is redundant

File: src/table/pagedTable/smoothPagedTable.tsx [hunk 3]
```diff
@@ -115,13 +114,32 @@ export function useSmoothPagedTable<T>(props: IUsePagedTableProps<T>) {
...
+    } catch (err) {
+      setError(err)
+    } finally {
+      setIsLoading(false)
+    }
+  }, [setLastPage, apiCall, setIsLoading, setRows, setPage])
+
   // load on component mount
   React.useEffect(() => {
+    console.log('useEffect')
```
Comment (valeriykundas) [inline src/table/pagedTable/smoothPagedTable.tsx:136]:
```suggestion
```

File: src/table/pagedTable/smoothPagedTable.tsx [hunk 4]
```diff
@@ -115,13 +114,32 @@ export function useSmoothPagedTable<T>(props: IUsePagedTableProps<T>) {
...
     if (autoLoad && !error) loadAPI()
   }, [autoLoad])
 
   // pagination controls callback
   const handleDidScrollToEnd = React.useCallback(() => {
+    console.log('handleDidScrollToEnd')
```
Comment (valeriykundas) [inline src/table/pagedTable/smoothPagedTable.tsx:142]:
```suggestion
```

**Review by gnudeb — APPROVED** [review]

File: src/table/pagedTable/smoothPagedTable.tsx [hunk 5]
```diff
@@ -115,6 +114,23 @@ export function useSmoothPagedTable<T>(props: IUsePagedTableProps<T>) {
     }
   }, [setLastPage, lastPage, apiCall, page, pageSize, queryParams, setIsLoading, setError])
 
+  const reloadFirstPage = React.useCallback(async() => {
+    try {
+      setIsLoading(true)
+      const res = await apiCall({ page: 1, pageSize, queryParams: queryParams })
+      setRows({
+        [1]: res.rows,
```
Comment (revmischa) [inline src/table/pagedTable/smoothPagedTable.tsx:122]: rather odd looking data structure to me - comment?

**Review by revmischa — APPROVED** [review]

File: src/table/pagedTable/smoothPagedTable.tsx [hunk 6]
```diff
@@ -115,6 +114,23 @@ export function useSmoothPagedTable<T>(props: IUsePagedTableProps<T>) {
     }
   }, [setLastPage, lastPage, apiCall, page, pageSize, queryParams, setIsLoading, setError])
 
+  const reloadFirstPage = React.useCallback(async() => {
```
[the remark this answers is not in the document]
Comment (d-spiridonov) [reply src/table/pagedTable/smoothPagedTable.tsx:122]: Right you are, I realized I can do it in a more elegant was, fixed, thanks!

**Review by rkyslyy — APPROVED** [review]

**Review by Dizzzmas — APPROVED** [review]

File: src/table/pagedTable/smoothPagedTable.tsx [hunk 7]
```diff
@@ -89,24 +89,28 @@ export function useSmoothPagedTable<T>(props: IUsePagedTableProps<T>) {
...
+      // if pageNumber is provided, we are refreshing this page, so the current page
+      // has to be set to the page that we are refreshing
+      if (pageNumber) {
+        setPage(pageNumber)
+      }
       // set newPage as current page
       // someone maybe scrolling like crazy so let's always remember last page
-
-      if (res.page) setPage(prevPage => Math.max(prevPage, res.page))
+      else if (res.page) setPage(prevPage => Math.max(prevPage, res.page))
```
Comment (revmischa) [inline src/table/pagedTable/smoothPagedTable.tsx:113]: try not to mix `{}`s and no `{}s` in a single if/else conditional

**Review by revmischa — APPROVED** [review]

**Review by adamszeptycki — APPROVED** [review]

**Review by valeriykundas — APPROVED** [review]

**Pull request merged**